### ISSUE TITLE
Improve sync UI/UX 

### ISSUE THREAD

**Issue by tobiasKaminsky** [nextcloud/android#3349]
From: https://github.com/nextcloud/android/pull/2886#issuecomment-446261587

> But it would be great if the meaning of the red cross is in some way explained to the user. Is it "broken file", "not accessible", "no download possible / server broken"... or "not up to date"?! I bet most users might assume one of the first explanations. Maybe you add a toast informing the user that there are outdated files that need to be synced manually while in a mobile network. And maybe a red/white clock would resemble "outdated" much better than a red cross. :-)
>
> Currently, the whole sync/offline/download feature is everything else than easy to understand for users.

This is how it currently looks like when a file is locally outdated / newer than server file. 
![2018-12-12-080613](https://user-images.githubusercontent.com/5836855/49852572-d011d300-fde4-11e8-9f0b-e4810b132572.png)

So we need a symbol for out of sync (local file is newer / older than server file).
Those are from https://material.io/tools/icons/
![image](https://user-images.githubusercontent.com/5836855/49852500-917c1880-fde4-11e8-875b-7ece6109aa5b.png) ![image](https://user-images.githubusercontent.com/5836855/49852513-9d67da80-fde4-11e8-99eb-8c918bc55212.png)

Aditionally, do yo have an idea how to explain this a bit more to the user?

@nextcloud/designers 
@AndyScherzinger 
@sehucke

**Comment by tobiasKaminsky** [comment]
Why don't we have downloaded? Above screenshot is the new situation, and a correct downloaded and up to date file, is marked with the green check mark.

**Comment by AndyScherzinger** [comment]
Yes, while before we had that for "make available offline" and for download(ed) we hat the yellow donload icon. So I guess we don't need the downloaded icon anymore if we have the sync status icon, right?

**Comment by tobiasKaminsky** [comment]
I think you remember our old icons:
https://github.com/nextcloud/android/blob/59cf6b9ba4965f38f2749f0624013daf0b83c946/src/main/java/com/owncloud/android/ui/adapter/OCFileListAdapter.java#L309-L332

New ones are:
![image](https://user-images.githubusercontent.com/5836855/49855762-6eeefd00-fdee-11e8-8710-aed3831a241d.png)

So we have
- synchronizing 
- conflict
- downloaded (this was the yellow arrow down icon, if I remember correctly)

**Comment by AndyScherzinger** [comment]
Yes I remembered the old icon :) While I think the up/download icons within a yellow circle could be used for that (or red, also fine with that).

Waiting to hear the @nextcloud/designers feedback ❤️

**Comment by sehucke** [comment]
IMHO, it would be over-simplified if you try to compress all the information in a few simple icons.

The cases that need to be visible and understandable for the user (currently and for future 2-way sync - just to think this through):
1. File exists on the server only
2. File exists on both client and server
    1. Local version and server version match
    2. Local version is newer than the server version
    3. Local version is older than the server version
    4. Local version and server version both changed (conflict!)
3. File exists on the client only
    1. New file was added on the client
    2. Server version was deleted

I would suggest to boil the annotated icons down to their central function: quick status overview for the user, i.e. draw the user's attention to things he must handle. And then provide means to make the user capable of resolving the situation.

What about four icons, for...
- ... local version and server version match (e.g. a green check mark),
- ... syncing in progress (e.g. rotating arrows on blue ground),
- ... outstanding background actions like sync up/down, delete etc. (e.g. a yellow exclamation mark),
- ... a sync conflict, when it comes to 2-way sync (e.g. a red cross).

In this case the user could easily distinguish by color:
1. *Green or no icon:* Nothing to do
2. *Blue:* Lean back and wait
3. *Yellow:* Do not care unless you need the file right now.
4. *Red:* Oh, I got a problem to solve (right now)

The last two cases may need further action by the user: What do you think about an "info" icon just left of the "share" icon (maybe colored like the annotated icon)? By clicking it the user would get detailed information about why there is a yellow or red annotation.
1. Newer version available on the server. Will be synced on WiFi.
2. Outstanding upload. Will be synced on WiFi.
3. Server version got deleted. Will be synced on WiFi.
3. File version conflict.

That message could then be extended with actions to resolve the situation
1. Sync now (even over mobile networks)
2. Conflict resolution
    1. Keep local version
    2. Keep server version
    3. keep both and auto-rename the oldest

And to make this more complete:
If the server is currently unreachable you could show a warning message on top of the folder list as this would be a general "problem". Or/And you could display the last time the server's folder was checked for changes beneath the current folder name at the top.

**Comment by jancborchardt** [comment]
> For many users it is a crucial requirement that they can sync on WiFi only.

If we sync on wifi only, then how do we know about the possible conflict? Getting the meta update from the server also incurs a data load, or do we see that as negligible?

Regarding your proposal:

> ... outstanding background actions like sync up/down, delete etc. (e.g. a yellow exclamation mark),

This should also be the blue circular arrows. Don’t see the need for the yellow exclamation mark there.

> ... a sync conflict, when it comes to 2-way sync (e.g. a red cross).

Right, when an actual _conflict_ happened with _different_ changes on mobile and server, we can show the red x. But not when the local file is older than the one on the server, as was shown in the screenshot.


> What do you think about an "info" icon just left of the "share" icon (maybe colored like the annotated icon)?

This is too much in general cases. If there is a sync conflict then yes we can go for this, that’s a good idea! :)

But for the normal case of just files having changed on the server, stuff should just work. If you downloaded a file previously, it is apparent that it’s of some importance to you.
